core/src/main/scala/org/bitcoins/core/crypto/MnemonicCode.scala

```diff
@@ -0,0 +1,247 @@
+package org.bitcoins.core.crypto
+
+import java.security.SecureRandom
+
+import org.bitcoins.core.protocol.NetworkElement
+import org.bitcoins.core.util.{CryptoUtil, Factory}
+import scodec.bits.{BitVector, ByteVector}
+
+import scala.annotation.tailrec
+import scala.io.Source
+
+sealed abstract class MnemonicCode extends NetworkElement {
+
+  private val VALID_LENGTHS = Vector(12, 15, 18, 21, 24)
```

Comment:
Scaladoc indicating what these valid lengths mean, I"m assuming they corresponding to the amount of words in a BIP39 seed?